File: c++/random_bot.cpp
#include <iostream>
#include <vector>
#include <random>
#include "api.hpp"

void random_bot() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dir_dist(0, 1);
    std::uniform_int_distribution<> action_dist(0, 1);

    Simulation sim("secondspace2025", "tiny_random", "random-cpp-bot", "bob@mail.com", true);
    json current_state = sim.get_initial_state();

    while (current_state["running"]) {
        std::vector<Command> commands;
        for (const auto& elevator : current_state["elevators"]) {
            commands.push_back({
                elevator["id"],
                dir_dist(gen) == 0,
                action_dist(gen) == 0
            });
        }
        current_state = sim.send(commands);
    }

    std::cout << "Score: " << current_state["score"] << std::endl;
    std::cout << "Replay URL: " << current_state["replay_url"] << std::endl;
}

int main() {
    random_bot();
    return 0;
}

File: java/RandomBot.java
import com.google.gson.JsonObject;
import com.google.gson.JsonArray;
import java.util.List;
import java.util.ArrayList;
import java.util.Random;

public class RandomBot {

    public static void main(String[] args) {
        randomBot();
    }

    public static void randomBot() {
        Api.Simulation simulation = new Api.Simulation(
                "secondspace2025",
                "tiny_random",
                "random-java-bot",
                "bob@mail.com",
                true
        );
        JsonObject currentState = simulation.initialState;
        Random random = new Random();

        while (currentState.get("running").getAsBoolean()) {
            List<Api.Command> commands = new ArrayList<>();
            JsonArray elevators = currentState.getAsJsonArray("elevators");
            for (int i = 0; i < elevators.size(); i++) {
                JsonObject elevator = elevators.get(i).getAsJsonObject();
                boolean direction = random.nextBoolean() ? Api.UP : Api.DOWN;
                boolean action = random.nextBoolean() ? Api.MOVE : Api.STOP;
                commands.add(new Api.Command(elevator.get("id").getAsString(), direction, action));
            }
            currentState = simulation.send(commands);
        }

        System.out.println("Score: " + currentState.get("score").getAsInt());
        System.out.println("Replay URL: " + currentState.get("replay_url").getAsString());
    }
}
